### PR TITLE
check jenkins good name & use name as id

### PR DESCRIPTION
**Pull request by runzexia**

### CODE REVIEW

File: pkg/service/projects/project.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package projects
+
+import (
+	"fmt"
+	"strings"
+)
+
+func checkJenkinsGoodName(name string) error {
```
Comment (magicsong) [inline pkg/service/projects/project.go]: use regex `^\w+$`

File: pkg/models/project.go [hunk 2]
```diff
@@ -43,7 +41,7 @@ type Project struct {
 
 func NewProject(name, description, creator, extra string) *Project {
 	return &Project{
-		ProjectId:   idutils.GetUuid(ProjectPrefix),
+		ProjectId:   name,
```
[the remark this answers is not in the document]
Comment (magicsong) [reply pkg/models/project.go]: front knows

File: pkg/service/projects/project.go (hunk 1 above)
Comment (magicsong) [reply pkg/service/projects/project.go]: https://github.com/kubernetes/kubernetes/blob/7f2d0c0f710617ef1f5eec4745b23e0d3f360037/pkg/util/validation.go#L26

File: pkg/models/project.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (runzexia) [reply pkg/models/project.go]: Modifying the Jenkins pipeline to k8s ns requires a lot of API path changes, not done in this PR

File: pkg/utils/validationutils/validation.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+func IsDNSLabel(value string) bool {
+	return IsDNS1123Label(value)
+}
+
+func IsDNS1123Label(value string) bool {
+	return len(value) <= dns1123LabelMaxLength && dns1123LabelRegexp.MatchString(value)
+}
+
+const dns1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
```
Comment (magicsong) [inline pkg/utils/validationutils/validation.go]: this regular exp is not corrected

**Pull request closed without merging**